lib/queue.rb

```diff
@@ -1,31 +1,57 @@
 class Queue
 
   def initialize
-    # @store = ...
-    raise NotImplementedError, "Not yet implemented"
+    @store = Array.new(100)
+    @front = -1
+    @back = -1
   end
 
   def enqueue(element)
-    raise NotImplementedError, "Not yet implemented"
+    if @front == -1 && @back == -1
+      @front = 0
+      @back = 0
+    end
+
+    @store[@back] = element
+    if (@back = (@back + 1) % @store.length) == @front
+      raise ArgumentError.new("Error: queue is full")
+    end
   end
 
   def dequeue
-    raise NotImplementedError, "Not yet implemented"
+    if @front == -1 && @back == -1
+      @front = 0
+      @back = 1
+    end
```

Comment:
If the queue is empty and you're removing an element you set the front to 0?

I'd suggest raising an error that the data structure is empty.